### PR TITLE
Add third person camera view

### PR DESCRIPTION
**Pull request by jeus0630**
image

## Doing
- Added new feature of third-person-camera-view 
- Added new button which control the camera POV.

## Notes
- In VR mode, it is only available with first person camera view.

### CODE REVIEW

File: src/components/player-info.js [hunk 1]
```diff
@@ -77,10 +77,10 @@ AFRAME.registerComponent("player-info", {
       let isSkinnedAvatar = false;
       modelEl.object3D.traverse(function(o) {
         if (o.isSkinnedMesh) {
-          const headlessMesh = createHeadlessModelForSkinnedMesh(o);
-          if (headlessMesh) {
+          const mesh = createModelForSkinnedMesh(o);
+          if (mesh) {
             isSkinnedAvatar = true;
-            o.parent.add(headlessMesh);
+            o.parent.add(mesh);
```
Comment (juunini) [inline src/components/player-info.js:83]: We(me and @jeus0630) need help this lines.

Default status is not shown head of avatar.
But, we don't know how to toggle avatar head.

So, we change here to avatar head always shows.
If someone helps 'how to toggle avatar head' we can fix here.

File: src/react-components/ui-root.js [hunk 2]
```diff
@@ -1561,6 +1562,23 @@ class UIRoot extends Component {
                       </>
                     )}
                     <ChatToolbarButtonContainer onClick={() => this.toggleSidebar("chat")} />
+                    {entered && (
+                      <ToolbarButton
+                        icon={<VRIcon />}
```
[the remark this answers is not in the document]
Comment (markusTraber) [reply src/react-components/ui-root.js:1567]: How about making an option in the preferences to toggle between 1st and 3rd person?

File: src/components/player-info.js (hunk 1 above)
Comment (markusTraber) [reply src/components/player-info.js:83]: I think you can do that with the camera layers. Have a look at: https://github.com/mozilla/hubs/blob/cb20d48f4acc7e810928a9b7b302bdab51589c65/src/systems/camera-system.js#L281-L282

I will probably implement that myself this way. Will tell you here if I have sth. to show.

**Comment by markusTraber** [comment]
@jeus0630 @juunini As I told, I did my own implementation which works with the layers. Here you can see the [comparison to the Mozilla Master](https://github.com/mozilla/hubs/compare/master...farvel-space:space-client:upstream-stage/master/FARM-513_3rd-person-view?expand=1).

I implemented an option to switch between 1st and 3rd person via the preferences. Here is the [branch itself](https://github.com/farvel-space/space-client/tree/upstream-stage/master/FARM-513_3rd-person-view).

All the interesting stuff for you is in camera-system.js. I am not sure about how I implemented it. The function "setMode" basically ignores calls to switch to 3rd person if the user is connected via VR.

**Comment by juunini** [comment]
@markusTraber Thanks :)
I'll tell @jeus0630

**Comment by piyush2greedy** [comment]
@juunini 
Can you please help. Where do we need to make code change in Hubs Code to see the third person view.

**Comment by jeus0630** [comment]
@piyush2greedy You can check out here https://github.com/mozilla/hubs/pull/5660

**Comment by h3d-longnsp** [comment]
@jeus0630  How to enable this feature? I see that it had been merged to `master` and also the develop branch was deleted but on my Hubs cloud instance there is no option to toggle first/third person view

**Pull request merged**